http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/resources/EncodeModeResource.java

```diff
@@ -0,0 +1,58 @@
+package io.quarkus.ts.http.restclient.reactive.resources;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.reactive.client.api.ClientMultipartForm;
+import org.jboss.resteasy.reactive.server.multipart.MultipartFormDataInput;
+
+@Path("/encode")
+public class EncodeModeResource {
+
+    private static final String TEXT = "text";
+    private static final String FILE = "file";
+    private static final String FILE_NAME = "example";
+    private static final Logger LOGGER = Logger.getLogger(EncodeModeResource.class);
+
+    @Inject
+    UploadFileResource uploadFileResource;
+
+    @POST
+    @Path("/upload")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response handleUploadBooks(MultipartFormDataInput formDataInput) {
+        var inputPartText = formDataInput.getValues().get(TEXT).stream().findAny().orElse(null);
+        if (inputPartText != null) {
+            try {
+                var inputPartFile = formDataInput.getValues().get(FILE).stream().findAny().orElse(null);
+                String fileContent = new String(inputPartFile.getFileItem().getInputStream().readAllBytes(),
+                        StandardCharsets.UTF_8);
+                String text = inputPartText.getValue();
+
+                ClientMultipartForm form = ClientMultipartForm.create()
+                        .stringFileUpload(FILE, FILE_NAME, fileContent, MediaType.TEXT_PLAIN)
+                        .attribute(TEXT, text, FILE_NAME);
+
+                Map<String, String> response = uploadFileResource.uploadBooks(form);
```

Comment:
Injecting resource kinda beat purpose of this whole test. Also even though the resource is a CDI bean, something tells me it is a bad practice. Never read it, so you don't have to believe me.

Reply:
fixed it, thanks.